### ISSUE TITLE
Fluent configuration support for CsvOutputFormatter

### ISSUE THREAD

**Issue by piotrkolodziej** [WebApiContrib/WebAPIContrib.Core#133]
Hello community,

I would very much like to add 'fluent' configuration support for CsvOutputFormatter. It would work similarly to FluentValidation or EF, and would look like:

Startup.cs

```
services
	.AddMvc()
	.AddCsvSerializerFormatters(options =>
		options.RegisterConfigsFromAssemblyContaining<BusinessProfile>();
	);
```

InvoiceFormatterConfig.cs - a config

```
	public class InvoiceFormatterConfig : CsvFormattingConfiguration<InvoiceModel>
	{
		public void Configure(CsvFormattingBuilder<InvoiceModel> builder)
		{
			builder
				.UseHeaders()
				.ForHeader("Name")
				.UseProperty(i => i.Name)
				.ForHeader("GrandTotal")
				.UseProperty(i => i.InvoiceDetails.GrandTotal);
		}
	}
```

CsvOutputFormatter would use this configuration and would be able to generate CSV based on it. Please let mne know what do you think. I am doing it anyway for my project, I just think it's something that would be far better than creating separate model type with primitive properties and using automapper to map entity type to our new model––which would exist only for the purpose of formatters. What do you think? Are you interested in my contribution?

ps: this could be also implemented for other formatter types.

Best Regards
Piotr Kolodziej

**Comment by damienbod** [comment]
@piotrkolodziej This would be cool. Would you add an extra extension method?

Greetings Damien

**Comment by piotrkolodziej** [comment]
@damienbod Let me work on a prototype for few days and reply to your question when I have more details.

**Comment by piotrkolodziej** [comment]
@damienbod what’s the procedure assuming I have a fork ready to share with you? This is my first contribution and also first time I am dealing with forks. Is it possible to create something similar to pull request so that you will be able to look over the changes?

**Comment by filipw** [comment]
Yes just push your changes to a new branch in your fork. You will then be able to open a PR from your fork into WebAPIContrib.Core `master`.

As soon as you push a branch with changes to your fork, and navigate to the root of WebApiContrib.Core repo, GH will automatically detect that you have some changes and offer a new button to open a PR.

Alternatively you can always trigger a PR from a fork branch using the following link: https://github.com/WebApiContrib/WebAPIContrib.Core/compare/master...piotrkolodziej:NAME_OF_YOUR_BRANCH

**Comment by piotrkolodziej** [comment]
Guys, just letting you know that prototype is ready but I will have a 1.5-2 weeks delay because I've got a sinus infection and will be changing location to India. Anyway don't close this topic, this is totally happening :-)

**Comment by piotrkolodziej** [comment]
@damienbod and @filipw , I've finished implementing (and testing) the fluent version of formatters. Let's keep in touch :-)
